pkg/infoschema/sieve.go

```diff
@@ -157,13 +156,10 @@ func (s *Sieve[K, V]) Get(key K) (value V, ok bool) {
 		var v V
 		failpoint.Return(v, false)
 	})
-
-	s.mu.RLock()
-	e, ok := s.items[key]
-	s.mu.RUnlock()
-
-	if ok {
-		e.visited.Store(true)
+	s.mu.Lock()
+	defer s.mu.Unlock()
```

Comment:
/hold

Do you have time to locate the root cause? Maybe we only need to increase the lock scope a bit

Reply:
Inside write lock protected code, it's

```
	if e, ok := s.items[key]; ok {
		e.value = value
		e.visited.Store(true)
		return
	}
```

rather than

```
	if e, ok := s.items[key]; ok {
		s.items[key] = &entry{
                         key: key,
                         value: value,
                }
	}
```

It's not an atomic  update ...

So when the reader only partly read locked, it still data race.

```
mu.RLock()
e := s.items[key]
mu.RUnlock()

// read e.value
```

Reply:
If we change to 

```
	if e, ok := s.items[key]; ok {
		s.items[key] = &entry{
                         key: key,
                         value: value,
                }
	}
```

It's allocating under write lock, potentially slower.

Overall, changing from Mutex to  RWMutex is not a carefully thought.
If we do need better performance here, we need better design and more time to thought.
Revert it first.